geocoder/geocoder.cpp

```diff
@@ -52,25 +71,29 @@ geocoder::Type NextType(geocoder::Type type)
 bool FindParent(vector<geocoder::Geocoder::Layer> const & layers,
                 geocoder::Hierarchy::Entry const & e)
 {
-  for (auto const & layer : layers)
+  CHECK(!layers.empty(), ());
+  auto const & layer = layers.back();
+  for (auto const * pe : layer.m_entries)
   {
-    for (auto const * pe : layer.m_entries)
-    {
-      // Note that the relationship is somewhat inverted: every ancestor
-      // is stored in the address but the nodes have no information
-      // about their children.
-      if (e.m_address[static_cast<size_t>(pe->m_type)] == pe->m_nameTokens)
-        return true;
-    }
+    // Note that the relationship is somewhat inverted: every ancestor
```

Comment:
The name of the function is misleading. HasPrent?

Reply:
HasParent is better, will fix